> That is weird. Does it tell you the error message when it fails?

```
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.1.3, pluggy-1.0.0 -- /usr/bin/python3.8
cachedir: .pytest_cache
rootdir: /tmp/bazel/sandbox/linux-sandbox/662/execroot/__main__/bazel-out/k8-fastbuild/bin/lte/gateway/python/magma/pipelined/tests/test_ifaces.runfiles/__main__
plugins: cov-3.0.0, flaky-3.7.0
collecting ... collected 0 items / 1 error

==================================== ERRORS ====================================
___ ERROR collecting lte/gateway/python/magma/pipelined/tests/test_ifaces.py ___
lte/gateway/python/magma/pipelined/tests/test_ifaces.py:16: in <module>
    from magma.pipelined.ifaces import get_mac_address
lte/gateway/python/magma/pipelined/ifaces.py:16: in <module>
    from magma.pipelined.metrics import NETWORK_IFACE_STATUS
lte/gateway/python/magma/pipelined/metrics.py:14: in <module>
    from prometheus_client import Counter, Gauge
../python_deps_pypi__prometheus_client/prometheus_client/__init__.py:6: in <module>
    from . import platform_collector
../python_deps_pypi__prometheus_client/prometheus_client/platform_collector.py:57: in <module>
    PLATFORM_COLLECTOR = PlatformCollector()
../python_deps_pypi__prometheus_client/prometheus_client/platform_collector.py:20: in __init__
    self._add_metric("python_info", "Python platform information", info)
E   NameError: name 'info' is not defined
=============================== warnings summary ===============================
../python_deps_pypi__prometheus_client/prometheus_client/decorator.py:81
  /tmp/bazel/sandbox/linux-sandbox/662/execroot/__main__/bazel-out/k8-fastbuild/bin/lte/gateway/python/magma/pipelined/tests/test_ifaces.runfiles/python_deps_pypi__prometheus_client/prometheus_client/decorator.py:81: DeprecationWarning: invalid escape sequence \s
    DEF = re.compile('\s*def\s*([_\w][_\w\d]*)\s*\(')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR lte/gateway/python/magma/pipelined/tests/test_ifaces.py - NameError: na...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
========================= 1 warning, 1 error in 0.12s ==========================
```

```
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.1.3, pluggy-1.0.0 -- /usr/bin/python3.8
cachedir: .pytest_cache
rootdir: /tmp/bazel/sandbox/linux-sandbox/1170/execroot/__main__/bazel-out/k8-fastbuild/bin/lte/gateway/python/magma/pipelined/tests/test_ifaces.runfiles/__main__
plugins: cov-3.0.0, flaky-3.7.0
collecting ... collected 0 items / 1 error

==================================== ERRORS ====================================
___ ERROR collecting lte/gateway/python/magma/pipelined/tests/test_ifaces.py ___
lte/gateway/python/magma/pipelined/tests/test_ifaces.py:16: in <module>
    from magma.pipelined.ifaces import get_mac_address
lte/gateway/python/magma/pipelined/ifaces.py:16: in <module>
    from magma.pipelined.metrics import NETWORK_IFACE_STATUS
lte/gateway/python/magma/pipelined/metrics.py:14: in <module>
    from prometheus_client import Counter, Gauge
../python_deps_pypi__prometheus_client/prometheus_client/__init__.py:4: in <module>
    from . import exposition
../python_deps_pypi__prometheus_client/prometheus_client/exposition.py:11: in <module>
    from wsgiref.simple_server import make_server, WSGIRequestHandler
/usr/lib/python3.8/wsgiref/simple_server.py:13: in <module>
    from http.server import BaseHTTPRequestHandler, HTTPServer
/usr/lib/python3.8/http/server.py:96: in <module>
    import mimetypes
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:971: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:914: in _find_spec
    ???
<frozen importlib._bootstrap_external>:1407: in find_spec
    ???
<frozen importlib._bootstrap_external>:1379: in _get_spec
    ???
<frozen importlib._bootstrap_external>:1506: in find_spec
    ???
<frozen importlib._bootstrap_external>:142: in _path_stat
    ???
E   ValueError: embedded null byte
=============================== warnings summary ===============================
../python_deps_pypi__prometheus_client/prometheus_client/decorator.py:81
  /tmp/bazel/sandbox/linux-sandbox/1170/execroot/__main__/bazel-out/k8-fastbuild/bin/lte/gateway/python/magma/pipelined/tests/test_ifaces.runfiles/python_deps_pypi__prometheus_client/prometheus_client/decorator.py:81: DeprecationWarning: invalid escape sequence \s
    DEF = re.compile('\s*def\s*([_\w][_\w\d]*)\s*\(')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR lte/gateway/python/magma/pipelined/tests/test_ifaces.py - ValueError: e...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
========================= 1 warning, 1 error in 0.14s ==========================
```

--- a/lte/gateway/python/magma/pipelined/BUILD.bazel
+++ b/lte/gateway/python/magma/pipelined/BUILD.bazel
@@ -34,6 +34,7 @@ py_binary(
     python_version = "PY3",
     tags = TAG_SERVICE,
     deps = [
+        ":ifaces",
         ":pipelined_lib",
         "//lte/gateway/python/magma/pipelined/app:he",
         "//lte/gateway/python/magma/pipelined/app:of_rest_server",
@@ -51,7 +52,6 @@ py_library(
     srcs = [
         "check_quota_server.py",
         "datapath_setup.py",
-        "ifaces.py",
     ],
     deps = [
         ":gtp_stats_collector",
@@ -247,9 +247,19 @@ py_library(
 py_library(
     name = "gw_mac_address",
     srcs = ["gw_mac_address.py"],
+    deps = [":ifaces"],
 )
 
 py_library(
     name = "pg_set_session_msg",
     srcs = ["pg_set_session_msg.py"],
+)
+
+py_library(
+    name = "ifaces",
+    srcs = ["ifaces.py"],
+    deps = [
+        requirement("netifaces"),
+        ":metrics",
+    ],
 )

--- a/lte/gateway/python/magma/pipelined/gw_mac_address.py
+++ b/lte/gateway/python/magma/pipelined/gw_mac_address.py
@@ -15,7 +15,8 @@ import ipaddress
 import logging
 
 from lte.protos.mobilityd_pb2 import IPAddress
-from scapy.arch import get_if_addr, get_if_hwaddr
+from magma.pipelined.ifaces import get_mac_address
+from scapy.arch import get_if_addr
 from scapy.data import ETH_P_ALL, ETHER_BROADCAST
 from scapy.error import Scapy_Exception
 from scapy.layers.inet6 import getmacbyip6
@@ -43,7 +44,7 @@ def _get_gw_mac_address_v4(ip: IPAddress, vlan: str, non_nat_arp_egress_port: st
             "sending arp via egress: %s",
             non_nat_arp_egress_port,
         )
-        eth_mac_src = get_if_hwaddr(non_nat_arp_egress_port)
+        eth_mac_src = get_mac_address(non_nat_arp_egress_port)
         psrc = "0.0.0.0"
         egress_port_ip = get_if_addr(non_nat_arp_egress_port)
         if egress_port_ip:

--- a/lte/gateway/python/magma/pipelined/ifaces.py
+++ b/lte/gateway/python/magma/pipelined/ifaces.py
@@ -29,3 +29,10 @@ def monitor_ifaces(iface_names):
             status = 1 if iface in active else 0
             NETWORK_IFACE_STATUS.labels(iface_name=iface).set(status)
         yield from asyncio.sleep(POLL_INTERVAL_SECONDS)
+
+
+def get_mac_address(interface_name: str) -> str:
+    if_addresses = netifaces.ifaddresses(interface_name)[netifaces.AF_LINK]
+    if not if_addresses or not if_addresses[0].get('addr'):
+        raise ValueError(f"No mac address found for interface {interface_name}")
+    return if_addresses[0]['addr']

--- a/lte/gateway/python/magma/pipelined/main.py
+++ b/lte/gateway/python/magma/pipelined/main.py
@@ -39,13 +39,12 @@ from magma.pipelined.gtp_stats_collector import (
     MIN_OVSDB_DUMP_POLLING_INTERVAL,
     GTPStatsCollector,
 )
-from magma.pipelined.ifaces import monitor_ifaces
+from magma.pipelined.ifaces import get_mac_address, monitor_ifaces
 from magma.pipelined.rpc_servicer import PipelinedRpcServicer
 from magma.pipelined.service_manager import ServiceManager
 from ryu import cfg
 from ryu.base.app_manager import AppManager
 from ryu.ofproto.ofproto_v1_4 import OFPP_LOCAL
-from scapy.arch import get_if_hwaddr
 
 
 def main():
@@ -121,9 +120,9 @@ def main():
     if 'virtual_mac' not in service.config:
         if service.config['dp_router_enabled']:
             up_iface_name = service.config.get('nat_iface', None)
-            mac_addr = get_if_hwaddr(up_iface_name)
+            mac_addr = get_mac_address(up_iface_name)
         else:
-            mac_addr = get_if_hwaddr(service.config.get('bridge_name'))
+            mac_addr = get_mac_address(service.config.get('bridge_name'))
 
         service.config['virtual_mac'] = mac_addr
 

--- a/lte/gateway/python/magma/pipelined/tests/BUILD.bazel
+++ b/lte/gateway/python/magma/pipelined/tests/BUILD.bazel
@@ -721,6 +721,14 @@ pytest_test(
 )
 
 pytest_test(
+    name = "test_ifaces",
+    size = "small",
+    srcs = ["test_ifaces.py"],
+    imports = [LTE_ROOT],
+    deps = ["//lte/gateway/python/magma/pipelined:ifaces"],
+)
+
+pytest_test(
     name = "test_tunnel_id_mapper",
     size = "small",
     srcs = ["test_tunnel_id_mapper.py"],

--- a/lte/gateway/python/magma/pipelined/tests/test_ifaces.py
+++ b/lte/gateway/python/magma/pipelined/tests/test_ifaces.py
@@ -1,0 +1,38 @@
+"""
+Copyright 2022 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import sys
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+# Prevent flakiness due to prometheus library import
+sys.modules["magma.pipelined.metrics"] = MagicMock()
+
+from magma.pipelined.ifaces import get_mac_address
+
+
+@patch("magma.pipelined.ifaces.netifaces")
+def test_get_mac_address(netifaces_mock):
+    netifaces_mock.AF_LINK = 13
+    netifaces_mock.ifaddresses.return_value = {netifaces_mock.AF_LINK: [{"addr": "00:11:22:33:44:55"}]}
+
+    assert get_mac_address("eth0") == "00:11:22:33:44:55"
+
+
+@patch("magma.pipelined.ifaces.netifaces")
+def test_get_mac_address_invalid(netifaces_mock):
+    netifaces_mock.AF_LINK = 13
+    netifaces_mock.ifaddresses.return_value = {netifaces_mock.AF_LINK: []}
+    with pytest.raises(ValueError):
+        get_mac_address("eth0")

